controllers/spacerequest/spacerequest_controller.go

```diff
@@ -500,6 +445,107 @@ func (r *Reconciler) ensureSecretForProvisionedNamespaces(ctx context.Context, m
 	return nil
 }
 
+func (r *Reconciler) ensureSecretForProvisionedNamespace(
+	ctx context.Context,
+	memberClusterWithSpaceRequest cluster.Cluster,
+	subSpaceTargetCluster cluster.Cluster,
+	spaceRequest *toolchainv1alpha1.SpaceRequest,
+	namespace toolchainv1alpha1.SpaceNamespace,
+) (*toolchainv1alpha1.NamespaceAccess, error) {
+	// check if secret reference exists in SpaceRequest's status
+	na := r.findNamespaceAccessForNamespace(spaceRequest, namespace)
+	if na == nil {
+		// create secret generating the name
+		return r.createProvisionedNamespaceAccessSecret(ctx, memberClusterWithSpaceRequest, subSpaceTargetCluster, spaceRequest, namespace, "")
+	}
+
+	// check if secret referenced in SpaceRequest's status exists
+	if err := r.searchExistingSecretRef(ctx, memberClusterWithSpaceRequest, spaceRequest, *na); err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, err
+		}
+
+		// create with already selected name if it's missing
+		return r.createProvisionedNamespaceAccessSecret(ctx, memberClusterWithSpaceRequest, subSpaceTargetCluster, spaceRequest, namespace, na.SecretRef)
+	}
+
+	return na, nil
+}
+
+func (r *Reconciler) createProvisionedNamespaceAccessSecret(
+	ctx context.Context,
+	memberClusterWithSpaceRequest cluster.Cluster,
+	subSpaceTargetCluster cluster.Cluster,
+	spaceRequest *toolchainv1alpha1.SpaceRequest,
+	namespace toolchainv1alpha1.SpaceNamespace,
+	secretName string,
+) (*toolchainv1alpha1.NamespaceAccess, error) {
+	// create the secret for this namespace
+	clientConfig, err := r.generateKubeConfig(ctx, subSpaceTargetCluster, namespace.Name)
+	if err != nil {
+		return nil, err
+	}
+	clientConfigFormatted, err := clientcmd.Write(*clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeConfigSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: spaceRequest.Name + "-", // it will be used only when Name is empty
+			Name:         secretName,
+			Namespace:    spaceRequest.Namespace,
+			Labels: runtimeclient.MatchingLabels{
+				toolchainv1alpha1.SpaceRequestLabelKey:                     spaceRequest.GetName(),
+				toolchainv1alpha1.SpaceRequestProvisionedNamespaceLabelKey: namespace.Name,
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+		StringData: map[string]string{
+			"kubeconfig": string(clientConfigFormatted),
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(spaceRequest, kubeConfigSecret, r.Scheme); err != nil {
+		return nil, errs.Wrap(err, "error setting controller reference for secret "+kubeConfigSecret.Name)
+	}
+	if err := memberClusterWithSpaceRequest.Client.Create(ctx, kubeConfigSecret); err != nil {
+		return nil, errs.Wrap(err, "error while creating secret")
+	}
+	log.FromContext(ctx).Info("Created Secret", "Name", kubeConfigSecret.Name)
```

Comment:
If it's the case when `GenerateName` is used, does this line log the secret's name or is `kubeConfigSecret.Name` empty?

Reply:
I'm almost sure that after the `Create` is executed the `Name` field will be populated even when using `GenerateName`.

Reply:
I knew that the created resource would have the `Name` field populated but I wasn't sure whether the `kubeConfigSecret` reference would have it by the time you logged it but looks like it does. 👍